src/EventStore.Core/Telemetry/GCSuspensionMetric.cs

```diff
@@ -0,0 +1,54 @@
+using System;
+using System.Diagnostics.Tracing;
+using EventStore.Core.Time;
+
+namespace EventStore.Core.Telemetry;
+
+public class GcSuspensionMetric : EventListener {
+	private const int GcKeyword = 0x0000001;
+	private readonly DurationMaxTracker _tracker;
+	private readonly IClock _clock;
+	private Instant? _started;
+
+	public GcSuspensionMetric(IClock clock, DurationMaxTracker tracker) {
+		_tracker = tracker;
+		_clock = clock;
+	}
+
+	protected override void OnEventSourceCreated(EventSource eventSource) {
+		if (eventSource.Name.Equals("Microsoft-Windows-DotNETRuntime")) {
+			EnableEvents(eventSource, EventLevel.Verbose, (EventKeywords)GcKeyword);
+		}
+	}
+
+	protected override void OnEventWritten(EventWrittenEventArgs eventData) {
+		if (_tracker == null || _clock == null)
+			return;
+
+		if (eventData.EventName is not "GCSuspendEEBegin_V1" && eventData.EventName is not "GCSuspendEEEnd_V1")
```

Comment:
probably add a default case to the switch instead of the additional check up here